packages/components/src/InputFile/InputFile.test.tsx

```diff
@@ -212,15 +222,43 @@ describe("Post Requests", () => {
           validator={pngFileValidator}
         />,
       );
-      const input = container.querySelector("input[type=file]");
+      const input = container.querySelector(
+        "input[type=file]",
+      ) as HTMLInputElement;
 
-      fireEvent.change(input, { target: { files: [testFile] } });
+      await userEvent.upload(input, testFile);
 
       await waitFor(() => {
         expect(container).toContainHTML("Only .png files are allowed");
       });
     });
   });
+
+  describe("when the number of file uploads exceeds maxFiles", () => {
+    it("shows the validation message and does not upload files", async () => {
+      const { container, getByText } = render(
+        <InputFile
+          getUploadParams={fetchUploadParams}
+          maxFilesValidation={{ maxFiles: 2, numberOfCurrentFiles: 0 }}
+          allowMultiple={true}
+        />,
+      );
+
+      const input = container.querySelector(
+        "input[type=file]",
+      ) as HTMLInputElement;
+
+      await userEvent.upload(input, [testFile, testFile, testFile]);
+
+      await waitFor(() => {
+        expect(
+          getByText("Cannot exceed a maximum of 2 files."),
+        ).toBeInTheDocument();
+      });
+
+      expect(axios.request).not.toHaveBeenCalled();
```

Comment:
Appreciate verifying the axios request doesn't fire instead of just checking the error message is visible